### PR TITLE
[WC-941]: Add enzyme free config for jest

### PR DESCRIPTION
**Pull request by iobuhov**
## Checklist

-   Contains unit tests ❌
-   Contains breaking changes ❌
-   Compatible with: MX 7️⃣, 8️⃣, 9️⃣
-   Did you update version and changelog? ✅ ❌
-   PR title properly formatted (`[XX-000]: description`)? ✅ ❌

## This PR contains

-   [x] Bug fix
-   [x] Feature
-   [ ] Refactor
-   [ ] Documentation
-   [ ] Other (describe)

## What is the purpose of this PR?
This PR is to bring alternative Jest config as current (default) has issues with react hooks.
Issue is caused by `jest-react-hooks-shallow` library that alter hooks behavior.
You can check it by just removing `setupFilesAfterEnv` from current config.

### Example test
```js
import { createElement, useEffect } from "react";
import { render } from "@testing-library/react";

function useMyHook(_props: any): void {
    console.log("(1) start useMyHook");
    useEffect(() => {
        console.log("(3) effect 1");
        return () => console.log("(5) cleanup effect 1");
    });
    useEffect(() => {
        console.log("(4) effect 2");
        return () => console.log("(6) cleanup effect 2");
    });
    console.log("(2) end useMyHook");
}

function Box(): null {
    useMyHook(null);
    return null;
}

describe("react hooks issue in jest env", () => {
    test("use effect demo", () => {
        render(<Box />);
    });
});
```

### Expected output:
```
 PASS  test/demo.spec.tsx
  ● Console

    console.log
      (1) start useMyHook

      at log (test/demo.spec.tsx:5:13)

    console.log
      (2) end useMyHook

      at log (test/demo.spec.tsx:14:13)

    console.log
      (3) effect 1

      at log (test/demo.spec.tsx:7:17)

    console.log
      (4) effect 2

      at log (test/demo.spec.tsx:11:17)

    console.log
      (5) cleanup effect 1

      at log (test/demo.spec.tsx:8:30)
          at Array.forEach (<anonymous>)

    console.log
      (6) cleanup effect 2

      at log (test/demo.spec.tsx:12:30)
          at Array.forEach (<anonymous>)
```

### Actual result:
```
 PASS  src/__tests__/demo.spec.tsx
  ● Console

    console.log
      (1) start useMyHook

      at useMyHook (__tests__/demo.spec.tsx:5:13)

    console.log
      (3) effect 1

      at __tests__/demo.spec.tsx:7:17

    console.log
      (4) effect 2

      at __tests__/demo.spec.tsx:11:17

    console.log
      (2) end useMyHook

      at useMyHook (__tests__/demo.spec.tsx:14:13)
```
## Relevant changes
A new jest config add.

## What should be covered while testing?

How tests execute useEffect hook

### CODE REVIEW

**Comment by jeffersonRibeiro** [comment]
Hi @iobuhov Could you please bump the pluggable-widgets version on both package.json and lock file?

**Review by jeffersonRibeiro — APPROVED** [review]

File: packages/pluggable-widgets-tools/test-config/jest.enzyme-free.config.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const { join } = require("path");
+
+const projectDir = process.cwd();
+
+module.exports = {
+    clearMocks: true,
+    testRunner: "jest-jasmine2",
+    rootDir: join(projectDir, "src"),
+    setupFilesAfterEnv: [join(__dirname, "test-index-no-enzyme.js")],
+    testMatch: ["<rootDir>/**/*.spec.{js,jsx,ts,tsx}"],
```
Comment (legio-vi-ferrata) [inline packages/pluggable-widgets-tools/test-config/jest.enzyme-free.config.js:10]: We might need to add this new file to the root `jest.config.js`.

File: packages/pluggable-widgets-tools/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## [Unreleased]
 
+### Added
+
+-   We added new `jest.enzyme-free.config.js` config file that don't include enzyme as dependency.
```
[the remark this answers is not in the document]
Comment (legio-vi-ferrata) [reply packages/pluggable-widgets-tools/CHANGELOG.md:11]: Let's also assign a JIRA issue to this PR.

**Review by legio-vi-ferrata — COMMENTED** [review]
First round of review is done.

File: packages/pluggable-widgets-tools/test-config/jest.enzyme-free.config.js (hunk 1 above)
Comment (iobuhov) [reply packages/pluggable-widgets-tools/test-config/jest.enzyme-free.config.js:10]: Can you explain a bit? I think I'm missing the context.

Comment (iobuhov) [reply packages/pluggable-widgets-tools/test-config/jest.enzyme-free.config.js:10]: Okay, some things now clear. In short - we don't need to move it to root. This file is for manual use. It needed only for those who don't need `enzyme` and removing `enzyme` form root config will be a breaking change.

Comment (legio-vi-ferrata) [reply packages/pluggable-widgets-tools/test-config/jest.enzyme-free.config.js:10]: 👍

**Review by legio-vi-ferrata — APPROVED** [review]

**Comment by jeffersonRibeiro** [comment]
@iobuhov Could you please resolve the conflicts?

**Pull request merged**